ReText/window.py

```diff
@@ -519,6 +520,107 @@ def dropEvent(e):
         self.tabWidget.dropEvent = dropEvent
         self.tabWidget.setTabBarAutoHide(globalSettings.tabBarAutoHide)
 
+        self._init_tabs_context_menu()
+
+    def _init_tabs_context_menu(self):
+        tabBar = self.tabWidget.tabBar()
+        tabBar.setContextMenuPolicy(Qt.ContextMenuPolicy.CustomContextMenu)
+        tabBar.customContextMenuRequested.connect(self._on_tab_context_menu_requested)
+
+        self._tabs_ctx_menu = QMenu(self)
+        self._tabs_ctx_menu_actions = {}
+
+        for action_type in TabActionTypes:
+            if action_type is not TabActionTypes.Unknown:
+                self._tabs_ctx_menu_actions[action_type] = self._tabs_ctx_menu.addAction(action_type.value)
+            if action_type is TabActionTypes.CopyFilePath:
+                self._tabs_ctx_menu.addSeparator()
+
+    def _on_tab_context_menu_requested(self, p):
+
+        clicked_tab_index = self.tabWidget.tabBar().tabAt(p)
+        if -1 == clicked_tab_index:
+            raise RuntimeWarning(f"No tab found at [{p.x()};{p.y()}].")
```

Comment:
You probably wanted to use [warnings.warn()](https://docs.python.org/3/library/warnings.html#warnings.warn) function instead of `raise`?